test_util/cluster_api.py

```diff
@@ -271,244 +273,29 @@ def get_url(self, node, path, port=None):
-                logging.info('Waiting for application to be deployed: '
-                             'Not all instances are healthy: {}'.format(repr(data)))
-                return None
-            else:
-                logging.info('Waiting for application to be deployed: {}'.format(repr(data)))
-                return None
-
-        try:
-            return _pool_for_marathon_app(app_definition['id'])
-        except retrying.RetryError:
-            raise Exception("Application deployment failed - operation was not "
-                            "completed in {} seconds.".format(timeout))
-
-    def destroy_marathon_app(self, app_name, timeout=120):
-        """Remove a marathon app
-
-        Abort the test if the removal was unsuccesful.
-
-        Args:
-            app_name: name of the applicatoin to remove
-            timeout: seconds to wait for destruction before failing test
-        """
-        @retrying.retry(wait_fixed=1000, stop_max_delay=timeout * 1000,
-                        retry_on_result=lambda ret: not ret,
-                        retry_on_exception=lambda x: False)
-        def _destroy_complete(deployment_id):
-            r = self.get('/marathon/v2/deployments', headers=self._marathon_req_headers())
-            assert r.ok
-
-            for deployment in r.json():
-                if deployment_id == deployment.get('id'):
-                    logging.info('Waiting for application to be destroyed')
-                    return False
-            logging.info('Application destroyed')
-            return True
-
-        r = self.delete('/marathon/v2/apps' + app_name, headers=self._marathon_req_headers())
-        assert r.ok
-
-        try:
-            _destroy_complete(r.json()['deploymentId'])
-        except retrying.RetryError:
-            raise Exception("Application destroy failed - operation was not "
-                            "completed in {} seconds.".format(timeout))
+    def get_client(self, path, user='default', default_headers={}):
+        return test_util.helpers.ApiClient(
+            cluster=self,
+            user=self._get_user(user),
+            api_base=path,
+            default_headers=default_headers)
 
-    @contextmanager
-    def marathon_deploy_and_cleanup(self, app_definition, timeout=120, check_health=True, ignore_failed_tasks=False):
-        yield self.deploy_marathon_app(
-            app_definition, timeout, check_health, ignore_failed_tasks)
-        self.destroy_marathon_app(app_definition['id'], timeout)
+    def marathon(self, user='default'):
+        return test_util.marathon_api.MarathonApi.get_client(
+            cluster=self,
+            user=self._get_user(user))
 
     def metronome_one_off(self, job_definition, timeout=300, ignore_failures=False):
         """Run a job on metronome and block until it returns success
         """
         job_id = job_definition['id']
+        metronome_api = self.get_client('/service/metronome/v1')
```

Comment:
I'd include the trailing slash here.

Also might be handy to make a generic helper that is `cluster.metronome` where the metronome client can be accessed rather than a one-off in this function.